### PR TITLE
[ML] always write prediction_[score|probability] for classification inference

### PR DESCRIPTION
**Pull request by benwtrent**
In order to unify model inference and analytics results we
need to write the same fields.

prediction_probability and prediction_score are now written
for inference calls against classification models.

### CODE REVIEW

**Comment by benwtrent** [comment]
@elasticmachine update branch

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationInferenceResults.java [hunk 1]
```diff
@@ -74,6 +75,11 @@ public ClassificationInferenceResults(StreamInput in) throws IOException {
         } else {
             this.predictionFieldType = PredictionFieldType.STRING;
         }
+        if (in.getVersion().onOrAfter(Version.V_7_9_0)) {
```
Comment (davidkyle) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationInferenceResults.java:78]: `V_7_10_0` surely?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/inference/TreeInferenceModel.java [hunk 2]
```diff
@@ -178,7 +178,7 @@ private InferenceResults buildResult(double[] value,
                     classificationProbability(value),
                     classificationLabels,
                     null,
-                    classificationConfig.getNumTopClasses(),
+                    classificationConfig.getNumTopClasses() == 0 ? 1 : classificationConfig.getNumTopClasses(),
```
Comment (davidkyle) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/inference/TreeInferenceModel.java:181]: Should `LangIdentNeuralNetwork` have the same logic?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationInferenceResults.java (hunk 1 above)
Comment (benwtrent) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationInferenceResults.java:78]: potentially, I am not sure if we want to backport it to 7.9 as java side inference was introduced in 7.9

Comment (benwtrent) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationInferenceResults.java:78]: @droberts195 what do you think?

Comment (droberts195) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationInferenceResults.java:78]: If it means the data in the destination index would have inconsistent fields without this fix in 7.9 when the fields were consistent in 7.8 then this is fixing a regression and should go into 7.9.

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**